c.html.markdown

```diff
@@ -366,13 +371,17 @@ int main (int argc, char** argv)
   printf("\n");
 
   // *****NOTES*****:
-  // Loops and Functions MUST have a body. If no body is needed:
+  // If the body of control expression is only one statement than parenthesises
+  // are optional, but often recommended:
+  for (jj = 0; jj < 5; jj++)
+      printf("%d\n", jj);
+
+  // Loops and if statements MUST have a body. If no body is needed:
   int i;
   for (i = 0; i <= 5; i++) {
-    ; // use semicolon to act as the body (null statement)
   }
   // Or
-  for (i = 0; i <= 5; i++);
+  for (i = 0; i <= 5; i++); // use semicolon to act as the body (null statement)
```

Comment:
Moving this comment is wrong, the `;` is a statement terminator on this line, not a body.

Reply:
Every `for` loop has four parts, one of which is the statement to be executed (aka the body of the for loop). The statement could either be a simple statement, such as a call to `printf` as above, a compound statement (aka block) that is a collection of statements enclosed in `{` and `}`, or the null statement, which is a statement that just doesn't do anything, which is just written as `;`. See [this section of the gnu C manual](https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#The-Null-Statement) for reference.

The previous way it was written, it was a null statement inside of a compound statement and the comment implied that the semicolon inside the curlies was necessary when it wasn't. In real life C, for loops that do nothing will have either `;` or `{}`. Although `{;}` as written here would've worked, I've never personally seen it and I don't think it is intuitive.

I certainly think what's going on could be better explained, but I also think that the null statement isn't something that is important enough to spend anything more than a passing mention (if that) on in this introduction to C.

Reply:
```suggestion
  for (i = 0; i <= 5; i++); // semicolon acts as the body (null statement)
```